File: backend/src/main/java/com/example/kolla/models/Minutes.java
package com.example.kolla.models;

import com.example.kolla.enums.MinutesStatus;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Minutes {
    private Long id;
    private Meeting meeting;

    @Builder.Default
    private MinutesStatus status = MinutesStatus.DRAFT;

    private String draftPdfPath;
    private String draftDocxPath;
    private String confirmedPdfPath;
    private String secretaryPdfPath;
    private String secretaryDocxPath;
    private String contentHtml;
    private LocalDateTime hostConfirmedAt;
    private String hostConfirmationHash;
    private LocalDateTime secretaryConfirmedAt;
    private LocalDateTime reminderSentAt;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/types/minutes.ts
/**
 * Minutes type definitions.
 * Requirements: 25.1–25.7
 */

// ─── Minutes status enum ──────────────────────────────────────────────────────

export type MinutesStatus = 'DRAFT' | 'HOST_CONFIRMED' | 'SECRETARY_CONFIRMED'

export type MinutesVersion = 'draft' | 'confirmed' | 'secretary'

export type MinutesFormat = 'pdf' | 'docx'

// ─── Minutes domain model ─────────────────────────────────────────────────────

export interface Minutes {
  id: number
  meetingId: number
  status: MinutesStatus
  contentHtml?: string
  draftPdfPath?: string
  draftDocxPath?: string
  confirmedPdfPath?: string
  secretaryPdfPath?: string
  secretaryDocxPath?: string
  draftAvailable?: boolean
  confirmedAvailable?: boolean
  secretaryAvailable?: boolean
  draftDocxAvailable?: boolean
  secretaryDocxAvailable?: boolean
  createdAt: string   // ISO 8601 UTC+7
  updatedAt: string
  confirmedAt?: string
  confirmedByName?: string
  secretaryEditedAt?: string
}

export interface MinutesConfirmationResponse {
  minutes: Minutes
  signedPdfFileName: string
  signedPdfContentType: string
  signedPdfBase64: string
  signedPdfSha256: string
}

// ─── Request bodies ───────────────────────────────────────────────────────────

export interface EditMinutesRequest {
  contentHtml: string
}
